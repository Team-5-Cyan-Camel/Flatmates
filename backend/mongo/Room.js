import mongoose from 'mongoose';

import {rosterSchema} from './Roster';
import {reminderSchema} from './Reminder';
import {expenseSchema} from './Expense';


const roomSchema = mongoose.Schema({
    _id: mongoose.Types.ObjectId,
    host: {
        type: mongoose.Schema.Types.ObjectId, 
        ref: 'User'
    },
    rosters: {
        type: [rosterSchema]
    },
    reminders: {
        type: [reminderSchema]
    },
    expenses: {
        type: [expenseSchema]
    },
    users: [{
        type: mongoose.Schema.Types.ObjectId, 
        ref: 'User'
    }]
});

export default mongoose.model('Room', roomSchema);
