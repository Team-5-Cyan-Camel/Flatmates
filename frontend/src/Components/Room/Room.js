import React from "react";
import axios from "axios";
import { useState, useEffect } from "react";

const Room = ({ update, setRoom }) => {
  // useEffect(() => {
  //   axios
  //     .get("./")
  //     .then((res) => {
  //       setRoom(res.data);
  //       // console.log(res.data.rosters);
  //     })
  //     .catch(function (error) {
  //       console.log(error);
  //     });
  // }, [update]);

  return <div></div>;
};

export default Room;
