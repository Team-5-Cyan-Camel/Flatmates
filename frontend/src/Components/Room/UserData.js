import React from "react";
import Card from "react-bootstrap/Card";
import "bootstrap/dist/css/bootstrap.min.css";
import Button from "react-bootstrap/Button";
import { FaTimes as Cross } from "react-icons/fa";
import axios from "axios";
import Form from "react-bootstrap/Form";
import { socket, SocketContext } from "../../Context/socketContext";

const UserData = ({ data, isHost, hostId }) => {
  const kickMember = (username) => {
    // console.log("KICLKK");
    // console.log(data);
    // console.log(username);
    const kickUser = {
      username: username,
    };
    axios
      .patch("/room/kick", kickUser)
      .then((res) => {
        // console.log(res.data);
      })
      .catch(function (error) {
        console.log(error);
      });
    socket.emit("update");
  };
  return (
    <>
      <Card
        id="Card-field"
        style={{
          alignItems: "center",
          justifyContent: "center",
          margin: "1em",
          width: "90%",
          backgroundColor: "#7e828b",
        }}
      >
        <Card.Header
          as="h5"
          id="Card-Header"
          className="text-center"
          style={{ width: "100%" }}
        >
          {" "}
          {data.name}{" "}
          {isHost && hostId !== data._id && (
            <Cross onClick={() => kickMember(data.username)} />
          )}
        </Card.Header>

        <Card.Body
          style={{
            alignItems: "start",
            justifyContent: "start",
            width: "100%",
          }}
        >
          <Form>
            <p>Number: {data.phoneNumber === "" ? "N/A" : data.phoneNumber}</p>
            <br></br>
            <p>Email: {data.email === "" ? "N/A" : data.email}</p>
            <br></br>
          </Form>
        </Card.Body>
      </Card>
    </>
  );
};

export default UserData;
