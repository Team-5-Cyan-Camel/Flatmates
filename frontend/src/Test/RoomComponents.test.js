import React from "react";
import renderer from "react-test-renderer";
import Room from "../Components/Room/Room";
import UserData from "../Components/Room/UserData";

// dummy function
function setIsHost() {
  return null;
}

let room = {
  users: [
    {
      _id: "6094f73787fe4e0ff89ef600",
      username: "UK",
      email: "",
      name: "Leon",
      phoneNumber: "",
    },
    {
      _id: "6094ff16dba18c4bcc3e376e",
      username: "coke",
      email: "",
      name: "Coco",
      phoneNumber: "",
    },
    {
      _id: "608674f890c6b43a80d04f8e",
      username: "loine",
      name: "Leon",
      phoneNumber: "021 0817 9478",
      email: "whomst@gmail.com",
    },
  ],
  _id: "6094f73d87fe4e0ff89ef601",
  host: "6094f73787fe4e0ff89ef600",
  rosters: [
    {
      assignedUsers: [
        {
          _id: "6094f73787fe4e0ff89ef600",
          username: "UK",
          name: "Leon",
        },
        {
          _id: "6094ff16dba18c4bcc3e376e",
          username: "coke",
          name: "Coco",
        },
        {
          _id: "608674f890c6b43a80d04f8e",
          username: "loine",
          name: "Leon",
        },
      ],
      _id: "6094fb79cd7f9941445dd4a1",
      title: "nice",
      tasks: [],
    },
    {
      assignedUsers: [
        {
          _id: "6094f73787fe4e0ff89ef600",
          username: "UK",
          name: "Leon",
        },
        {
          _id: "6094ff16dba18c4bcc3e376e",
          username: "coke",
          name: "Coco",
        },
        {
          _id: "608674f890c6b43a80d04f8e",
          username: "loine",
          name: "Leon",
        },
      ],
      _id: "6096785e44457b3fbca1f9b2",
      title: "nice",
      tasks: [],
    },
    {
      assignedUsers: [
        {
          _id: "6094f73787fe4e0ff89ef600",
          username: "UK",
          name: "Leon",
        },
        {
          _id: "6094ff16dba18c4bcc3e376e",
          username: "coke",
          name: "Coco",
        },
        {
          _id: "608674f890c6b43a80d04f8e",
          username: "loine",
          name: "Leon",
        },
      ],
      _id: "6096787044457b3fbca1f9b4",
      title: "mine",
      tasks: [],
    },
  ],
  reminders: [],
  expenses: [],
  __v: 16,
};

test("Room display renders correctly", () => {
  const tree = renderer

    .create(
      <Room
        update={null}
        room={room}
        hostId={room.host}
        setIsHost={setIsHost}
      />
    )
    .toJSON();
  expect(tree).toMatchSnapshot();
});

test("User info display renders correctly", () => {
  const firstUser = room.users[0];
  const tree = renderer
    .create(
      <UserData
        data={firstUser}
        hostId={"6094f73787fe4e0ff89ef600"}
        isHost={true}
      />
    )
    .toJSON();
  expect(tree).toMatchSnapshot();
});
