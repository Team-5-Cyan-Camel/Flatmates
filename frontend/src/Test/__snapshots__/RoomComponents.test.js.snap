// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Room display renders correctly 1`] = `
<div
  className="card"
  id="Card-field"
  style={
    Object {
      "alignItems": "center",
      "justifyContent": "center",
    }
  }
>
  <h5
    className="text-center card-header"
    id="Card-Header"
    style={
      Object {
        "width": "100%",
      }
    }
  >
     
    Members
  </h5>
  <div
    className="card-body"
    style={
      Object {
        "alignItems": "center",
        "display": "Flex",
        "justifyContent": "center",
        "width": "90%",
      }
    }
  />
</div>
`;

exports[`User info display renders correctly 1`] = `
<div
  className="card"
  id="Card-field"
  style={
    Object {
      "alignItems": "center",
      "justifyContent": "center",
    }
  }
>
  <h5
    className="text-center card-header"
    id="Card-Header"
    style={
      Object {
        "width": "100%",
      }
    }
  >
     
    Leon
     
  </h5>
  <div
    className="card-body"
    style={
      Object {
        "alignItems": "center",
        "display": "Flex",
        "justifyContent": "center",
        "width": "90%",
      }
    }
  >
    <form
      className=""
    >
      <p>
        Number: 
        N/A
      </p>
      <br />
      <p>
        Email: 
        N/A
      </p>
      <br />
    </form>
  </div>
</div>
`;
